### ISSUE TITLE
BUG: ewm not working properly with times

### ISSUE THREAD

**Issue by deepers** [pandas-dev/pandas#40966]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample

```python
import numpy as np
import pandas as pd

print('Example 1')

df1 = pd.DataFrame(dict(x=[1, 2],
                        t=[np.datetime64(1, 's'), np.datetime64(3, 's')]))

ewm1 = df1.ewm(halflife='1s', times='t').mean()['x']
ewm2 = df1.ewm(halflife='1s', times='t')['x'].mean()

print(df1, ewm1, ewm2, sep=2 * '\n')

print('\n\nExample 2')

df2 = pd.concat([df1, df1], keys=['a', 'b'], names=['group', None])
ewm3 = (df2
        .groupby('group', group_keys=False)
        .ewm(halflife='1s', times='t')
        .mean())

print(df2, ewm3, sep=2 * '\n')
```

#### Problem description

This code computes a rolling EWM of a two element time series `[1, 2]` with a one second half life, assuming two seconds between the data points. You get different answers depending on how you do the computation. The output I get is the following.

```
Example 1
   x                   t
0  1 1970-01-01 00:00:01
1  2 1970-01-01 00:00:03

0    1.0
1    1.8
Name: x, dtype: float64

0    1.0
1    2.0
Name: x, dtype: float64


Example 2
         x                   t
group                         
a     0  1 1970-01-01 00:00:01
      1  2 1970-01-01 00:00:03
b     0  1 1970-01-01 00:00:01
      1  2 1970-01-01 00:00:03

           x
group       
a     0  1.0
      1  2.0
b     0  1.0
      1  2.0
```
I would have thought that `ewm1` and `ewm2` should the same, but to me it looks like `ewm2` is incorrect. Also when you combine `ewm()` with `groupby()`, it looks like you get the same wrong answer in `ewm3` as in `ewm2`.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
python           : 3.8.8.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.9.0-15-amd64
Version          : #1 SMP Debian 4.9.258-1 (2021-03-08)
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : en_CA.UTF-8
LOCALE           : en_CA.UTF-8

pandas           : 1.2.4
numpy            : 1.20.2
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 54.1.2
Cython           : None
pytest           : 6.2.3
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 1.3.9
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.8.6 (dt dec pq3 ext lo64)
jinja2           : 2.11.3
IPython          : 7.22.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : 0.5.0
gcsfs            : None
matplotlib       : 3.4.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 3.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : 0.53.1
</details>

**Comment by stevenschaerer** [comment]
@mroeschke 

Saw this bug report by chance. This is already fixed on master.

```
Example 1
   x                   t
0  1 1970-01-01 00:00:01
1  2 1970-01-01 00:00:03

0    1.0
1    1.8
Name: x, dtype: float64

0    1.0
1    1.8
Name: x, dtype: float64


Example 2
         x                   t
group                         
a     0  1 1970-01-01 00:00:01
      1  2 1970-01-01 00:00:03
b     0  1 1970-01-01 00:00:01
      1  2 1970-01-01 00:00:03

                 x
group group       
a     a     0  1.0
            1  1.8
b     b     0  1.0
            1  1.8

```

**Comment by jakehsiao** [comment]
take this, even not sure whether can solve it or not.

**Comment by deepers** [comment]
@stevenschaerer Thanks for checking on master.

However, one thing I noticed in your output for `ewm3` is that `group_by()` doesn't seem to respect the `group_keys=False` argument.

**Comment by mroeschke** [comment]
Thanks for the report.

The ``__getitem__`` behavior looks be tested and will be fixed in the 1.3 release

https://github.com/pandas-dev/pandas/blob/04f9a4b10df91e6f4838dd8e1e41bd97141b6332/pandas/tests/window/test_ewm.py#L145

The `group_keys` docs (https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.groupby.html) state: 

>    When calling apply, add group keys to index to identify pieces.

therefore, `group_keys` should only impact `groupby(...).apply(...)`.

Since everything appears covered, going to close this issue.
